modules/social_features/social_activity/modules/social_activity_filter/src/Plugin/views/filter/ActivityFilterTags.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Drupal\social_activity_filter\Plugin\views\filter;
+
+use Drupal\views\Plugin\views\filter\FilterPluginBase;
+use Drupal\views\Views;
+
+/**
+ * Filters activity by the taxonomy tags in the stream block.
+ *
+ * @ingroup views_filter_handlers
+ *
+ * @ViewsFilter("activity_filter_tags")
+ */
+class ActivityFilterTags extends FilterPluginBase {
+
+  /**
+   * Not exposable.
+   */
+  public function canExpose() {
+    return FALSE;
+  }
+
+  /**
+   * Filters out activity items by the taxonomy tags.
+   */
+  public function query() {
+    $tags = $this->view->filter_tags;
+    $taxonomy_field = $this->view->filter_vocabulary;
+    $taxonomy_table = "node__{$taxonomy_field}";
+
+    // @todo: replace it!
```

Comment:
@volodymyr-sydor-lemberg As you noted yourself this need to be improved, with using DI.

Reply:
done